### PR TITLE
Support Python 3.10

### PR DESCRIPTION
**Pull request by jiasli**
Supporting Python 3.10 in Knack is the prerequisite for 

- https://github.com/Azure/azure-cli/issues/19857

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 argcomplete==1.12.2
 colorama==0.4.4
-flake8==3.8.4
+flake8==4.0.1
 jmespath==0.10.0
 Pygments==2.8.1
-pylint==2.7.2
-pytest==6.2.2
+pylint==2.11.1
+pytest==6.2.5
```
Comment (jiasli) [inline requirements.txt:7]: Old `pytest` fails in Python 3.10: https://github.com/pytest-dev/pytest/issues/8546#issuecomment-820232967

```
=============================================== short test summary info ===============================================
ERROR tests/test_cli_scenarios.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_command_registration.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_command_with_configured_defaults.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_completion.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_config.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_deprecation.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_experimental.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_help.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_introspection.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_log.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_output.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_parser.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_preview.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_prompting.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_query.py - TypeError: required field "lineno" missing from alias
ERROR tests/test_util.py - TypeError: required field "lineno" missing from alias
```

File: knack/config.py [hunk 2]
```diff
@@ -220,7 +222,7 @@ def getboolean(self, section, option):
 
     def set(self, config):
         ensure_dir(self.config_dir)
-        with open(self.config_path, 'w') as configfile:
+        with open(self.config_path, 'w', encoding=CONFIG_FILE_ENCODING) as configfile:
```
Comment (jiasli) [inline knack/config.py:225]: `pylint` warns:

> knack\config.py:225:13: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)

See https://pylint.pycqa.org/en/latest/technical_reference/features.html#stdlib-checker-messages

File: requirements.txt [hunk 3]
```diff
@@ -1,10 +1,10 @@
 argcomplete==1.12.2
 colorama==0.4.4
-flake8==3.8.4
+flake8==4.0.1
 jmespath==0.10.0
 Pygments==2.8.1
-pylint==2.7.2
-pytest==6.2.2
+pylint==2.11.1
```
Comment (jiasli) [inline requirements.txt:6]: New `pylint` is more strict on grammar. We need to fix these additional check failures.

File: .pylintrc [hunk 4]
```diff
@@ -10,7 +10,8 @@
 # R0401 cyclic-import
 # R0205 useless-object-inheritance
 # R1717 consider-using-dict-comprehension
-disable=W0511,C0111,C0103,C0415,I0011,R0913,R0903,R0401,R0205,R1717,useless-suppression
+disable=W0511,C0111,C0103,C0415,I0011,R0913,R0903,R0401,R0205,R1717,useless-suppression,
+    consider-using-f-string
```
Comment (jiasli) [inline .pylintrc:14]: There are countless places where `str.format` is used. We should allow it.

**Review by evelyn-ys — APPROVED** [review]

**Review by wangzelin007 — APPROVED** [review]
LGTM

**Pull request merged**